posts/2021-09-03-microprofile-21009.adoc

```diff
@@ -0,0 +1,266 @@
+
+To get started with and learn more about MicroProfile Health, check out our link:https://openliberty.io/guides/microprofile-health.html[MicroProfile Health guide].
+
+[#tbasic]
+== TBASIC Logging Format
+
+A new logging format, `TBASIC`, is added to match the Websphere Application Server basic logging format. This new format option can be used for `consoleFormat`, `messageFormat`, or `traceFormat`. `TBASIC` format matches the existing `BASIC` format for `traceFormat` (so now either `TBASIC` or `BASIC` can be used to refer to this format for `traceFormat`). Using this new format option will allow users that utilize log parsers that work with the Websphere Application Server basic format to use the same parsers with Open Liberty logs.
+
+The new options can be used in the `bootstrap.properties` file:
+
+[source, java]
+----
+com.ibm.ws.logging.message.format=tbasic
+com.ibm.ws.logging.console.format=tbasic
+com.ibm.ws.logging.trace.format=tbasic
+----
+
+You can also change the format by editing the `server.env` file and adding the following lines:
+
+[source, java]
+----
+WLP_LOGGING_MESSAGE_FORMAT=TBASIC
+WLP_LOGGING_CONSOLE_FORMAT=TBASIC
+----
+
+
+TBASIC Logs Example:
+
+[source, java]
+----
+[24/03/21 15:04:10:331 EDT] 00000001 FrameworkMana A   CWWKE0001I: The server defaultServer has been launched.
+[24/03/21 15:04:11:338 EDT] 00000001 FrameworkMana I   CWWKE0002I: The kernel started after 1.177 seconds
+[24/03/21 15:04:11:465 EDT] 0000003e FeatureManage I   CWWKF0007I: Feature update started.
+[24/03/21 15:04:11:635 EDT] 00000033 DropinMonitor A   CWWKZ0058I: Monitoring dropins for applications.
+----
+
+
+[#activeEndpoints]
+== New method to determine active endpoints
+
+The `ServerEndpointControlMBean` can currently be used to pause and resume endpoints, including `HTTP` and messaging ports. It can also determine if an endpoint is currently paused. However, there is no way to determine programmatically if an endpoint exists and is active. This update provides a new method on the `mbean`, `isActive`, that will determine if an endpoint exists, has started, and is not paused.
+A new method was added to the `ServerEndpointControlMBean`:
+
+`public boolean isActive(String targets);`
+
+The `targets` parameter is a comma separated list of endpoint names. 
+
+This mirrors the existing `isPaused` method. It will return true only if all of the endpoints listed exist, are started, and are not paused.
+
+[source, java]
+----
+// Check if the defaultHttpEndpoint is active
+boolean isEndpointActive = mbean.isActive("defaultHttpEndpoint");
+----
+
+For more information about `ServerEndpointControlMBean` methods, refer to the link:https://www.ibm.com/docs/en/was-liberty/nd?topic=SSAW57_liberty/com.ibm.websphere.javadoc.liberty.doc/com.ibm.websphere.appserver.api.kernel.service_1.0-javadoc/com/ibm/websphere/kernel/server/ServerEndpointControlMBean.html[API documentation].
+
+
+[#transaction]
```

Comment:
Just like you moved this section to the second item in the list at the top, this section needs to be moved to follow the MP section